docs/src/ideal.md

```diff
@@ -0,0 +1,182 @@
+```@meta
+CurrentModule = AbstractAlgebra
+DocTestSetup = quote
+    using AbstractAlgebra
+end
+```
+
+# Ideal functionality
+
+AbstractAlgebra.jl provides a module, implemented in `src/generic/Ideal.jl` for
+ideals of a Euclidean domain (assuming the existence of a `gcdx` function)
+or of a univariate or multivariate polynomial ring
+over the integers. Univariate and multivariate polynomial rings over other
```

Comment:
Out of curiosity: the code seems to be mostly generic (judging by the type), so where does this restriction come in? Perhaps the use of `max` in the code is part of the reason? Or something else?

Reply:
One needs heuristics for the "size" of polynomials. They are pretty subtle and depend very much on the ring. Over the integers things like the number of bits of the coefficients (appropriately weighted) are used.